lib/ain-evm/src/backend.rs

```diff
@@ -249,8 +249,12 @@ impl EVMBackend {
             }
 
             if let Some(code) = code {
-                self.storage
-                    .put_code(self.vicinity.block_number, account.code_hash, code)?;
+                self.storage.put_code(
+                    self.vicinity.block_number,
```

Comment:
Can use self.block_number()

Reply:
This isnt possible because of the mutable borrow that occurs in the drain method which prevents us from calling the method which creates another immutable borrow....